### PR TITLE
min/max support for SymInt/Floats, finish as_strided/scatter/squeeze() backward symint support

### PR DESCRIPTION
**Pull request by bdhirsh**
This PR shouldn't matter too much, but I figured I'd land it instead of deleting. `PySymInt.min/max` are technically broken today, and this fixes them - but it doesn't matter (yet) because nobody is calling `min()` / `max()` on symints from python (they all happen using `std::min/max` in C++, which desugar to lt / gt calls).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86611
* #86610
* __->__ #86609

### CODE REVIEW

**Comment by albanD** [comment]
This is already in master now

**Comment by bdhirsh** [comment]
It looks like https://github.com/pytorch/pytorch/pull/86643 hasn't landed yet?

(Separately, seems like CI didn't kick off after I updated this PR for some reason)

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -231,8 +232,7 @@ def _make_magic(method, func, py_type):
 
     def magic_impl(self, other):
         if method in ["min", "max"]:
-            # op = getattr(builtins, method)
-            return self
+            op = getattr(builtins, method)
```
Comment (bdhirsh) [inline torch/fx/experimental/symbolic_shapes.py:235]: @albanD I could delete this PR, but I noticed this - probably still worth landing?

**Review by albanD — APPROVED** [review]
SGTM

File: torch/fx/experimental/symbolic_shapes.py [hunk 2]
```diff
@@ -255,12 +255,11 @@ def magic_impl(self, other):
             return py_type(out, self.shape_env)
 
     def unary_magic_impl(self):
+        if method in ["ceil", "floor"]:
```
[the remark this answers is not in the document]
Comment (bdhirsh) [reply torch/fx/experimental/symbolic_shapes.py:258]: yep - i'll remove this

**Pull request closed without merging**